synapse/storage/databases/main/relations.py

```diff
@@ -514,16 +514,22 @@ def _get_thread_summaries_txn(
                         AND parent.room_id = child.room_id
                     WHERE
                         %s
-                        AND relation_type = ?
+                        AND %s
                     ORDER BY child.topological_ordering DESC, child.stream_ordering DESC
                 """
 
             clause, args = make_in_list_sql_clause(
                 txn.database_engine, "relates_to_id", event_ids
             )
+
             args.append(RelationTypes.THREAD)
+            if self._msc3440_enabled:
+                relations_clause = "(relation_type = ? OR relation_type = ?)"
+                args.append(RelationTypes.UNSTABLE_THREAD)
+            else:
+                relations_clause = "relation_type = ?"
```

Comment:
I'd find this easier to follow:

```suggestion
            if self._msc3440_enabled:
                relations_clause = "(relation_type = ? OR relation_type = ?)"
                args.extend((RelationTypes.THREAD, RelationTypes.UNSTABLE_THREAD))
            else:
                relations_clause = "relation_type = ?"
                args.append(RelationTypes.THREAD)
```

likewise below